BUG: Fix Series.is_unique with single occurrence of NaN

jschendel opened the pull request:
- [X] closes #25180
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

jschendel committed:
BUG: Fix Series.is_unique with single occurrence of NaN

jreback commented on pandas/tests/series/test_duplicates.py:
```diff
@@ -59,12 +59,19 @@ def test_unique_data_ownership():
-    assert s.is_unique is False
-    s = Series(np.arange(1000))
-    assert s.is_unique is True
+@pytest.mark.parametrize('data, expected', [
```
do we have sufficient tests on the Index side for this? (e.g. your parameterization is good, maybe add for the index tests)/

jreback commented:
lgtm. if we have sufficient tests, merge on green, otherwise if you can add.

jschendel committed:
small change for consistency

jschendel committed:
index tests

jschendel commented:
@jreback : Added index tests.  Didn't see many `is_unique` tests for index; some indexes tested it but other didn't appear to.  Added a test to the base class to test `is_unique` in general, as well as the `NaN` specific issue.  Removed some tests that were made redundant by the common test.

jreback commented:
thanks!
